File: Bajra/email.cpp
#include <iostream>
#include <string>
using namespace std;

bool validateEmail(const string &email) {
    size_t index = email.find('@');
    if (index == string::npos) return false; // @ check

    if (email.size() < 4 || email.substr(email.size() - 4) != ".com")
        return false;

    string localPart = email.substr(0, index);
    int count = 0; // letters before @
    bool haveNumber = false;

    for (char c : localPart) {
        if ((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z')) {
            count++;
        }
        if (c >= '0' && c <= '9') {
            haveNumber = true;
        }
    }

    if (count < 6) return false;
    if (!haveNumber) return false;

    return true;
}

int main() {
    cout << boolalpha << validateEmail("sub32@gmail.com") << endl; // false
    cout << boolalpha << validateEmail("subhash32@gmail.com") << endl; // true
}

File: email.js
function validateEmail(email){
    let index =  email.indexOf('@');
    if(!index) return false;
    // return index;
    if(!email.endsWith(".com")) return false;
    let localPart = email.slice(0,index);
    let count=0; //count of letters before @
    let haveNumber = false;

    for(let char of localPart){
        if((char>='a' && char<='z') || (char >='A' && char<='Z')){
            count++;
        }

        if(char>=0 && char<=9){
            haveNumber = true;
        }
    }
    if(count<6) return false;
    if(!haveNumber) return false;

    return true;
}
console.log(validateEmail("sub32@gmail.com"))